astroplan/constraints.py

```diff
@@ -0,0 +1,614 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Specify and constraints to determine which targets are observable for
+an observer.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+# Standard library
+from abc import ABCMeta, abstractmethod
+import datetime
+
+# Third-party
+from astropy.time import Time
+import astropy.units as u
+from astropy.coordinates import get_sun, Angle
+from astropy.coordinates.angle_utilities import angular_separation
+from astropy import table
+import numpy as np
+
+# Package
+from .moon import get_moon, moon_illumination
+from .utils import time_grid_from_range
+
+__all__ = ["AltitudeConstraint", "AirmassConstraint", "AtNightConstraint",
+           "is_observable", "is_always_observable", "time_grid_from_range",
+           "SunSeparationConstraint", "MoonSeparationConstraint",
+           "MoonIlluminationConstraint", "LocalTimeConstraint", "Constraint"]
+
+@abstractmethod
+class Constraint(object):
+    """
+    Abstract class for objects defining observational constraints.
+    """
+    __metaclass__ = ABCMeta
+
+    def __call__(self, observer, targets, times=None,
+                 time_range=None, time_grid_resolution=0.5*u.hour):
+
+        if times is None and time_range is not None:
+            times = time_grid_from_range(time_range,
+                                         time_resolution=time_grid_resolution)
+        elif not isinstance(times, Time):
+            times = Time(times)
+
+        if times.isscalar:
+            times = Time([times])
+
+        cons = self.compute_constraint(times, observer, targets)
+        return cons
+
+    def compute_constraint(self, times, observer, targets):
+        # Should be implemented on each subclass of Constraint
+        raise NotImplementedError
+
+    def _get_altaz(self, times, observer, targets,
```

Comment:
This method is not accessing `self`. -> Make it a `staticmethod` or standalone function?